### PR TITLE
PCP-230: Fix AWS Cluster Tags Deletion & Re-addition issue

### PR DESCRIPTION
**Pull request by vishu2498**
**What type of PR is this?** /kind bug

**What this PR does / why we need it**:

Fixing AWS Subnet Tags issue where tags are not getting deleted/updated on static placement cluster deletion

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # PCP-230

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/523#" title="Author self-approved">vishu2498</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-v1beta1/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/cloud/services/network/subnets.go [hunk 1]
```diff
@@ -275,8 +276,48 @@ func (s *Service) getDefaultSubnets() (infrav1.Subnets, error) {
...
+		// If the tag is found, it will be deleted. If not found, an error will be returned.
+		// The fetching of tags is done in the same way tags are described using AWS CLI.
+		existing, err := s.describeSubnets()
+		if err != nil {
+			return err
+		}
+
+		for i := range existing.Subnets {
+
+			resources := []string{"resource-type", "key", "value"}
```
Comment (sadysnaat) [inline pkg/cloud/services/network/subnets.go:295]: maybe use these values directly, let's consult how other filter inputs are genertated

Comment (vishu2498) [reply pkg/cloud/services/network/subnets.go]: Changed accordingly

**Review by sadysnaat — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/523#pullrequestreview-1214855247" title="Approved">sadysnaat</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/523#" title="Author self-approved">vishu2498</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-v1beta1/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**